Assets/SteamVR_Unity_Toolkit/Scripts/Controls/3D/VRTK_Chest.cs

```diff
@@ -0,0 +1,227 @@
+﻿namespace VRTK
+{
+    using UnityEngine;
+
+    public class VRTK_Chest : VRTK_Control
+    {
+        public enum Direction
+        {
+            autodetect, x, z // no support for y at this point in time
+        }
+
+        public Direction direction = Direction.autodetect;
+        private float min = 0f;
+        public float max = 160f;
+        private float stepSize = 1f;
+
+        public GameObject lid;
+        public GameObject body;
+        public GameObject handle;
+
+        private Rigidbody handleRb;
+        private FixedJoint handleFj;
+        private VRTK_InteractableObject handleIo;
+        private Rigidbody lidRb;
+        private HingeJoint lidHj;
+        private Rigidbody bodyRb;
+
+        private Direction finalDirection;
+        private float subDirection = 1; // positive or negative can be determined automatically since handle dictates that
+        private bool lidHjCreated;
+
+        public override void OnDrawGizmos()
+        {
+            base.OnDrawGizmos();
+
+            // show opening direction
+            Bounds handleBounds = Utilities.GetBounds(handle.transform, handle.transform);
+            float length = handleBounds.extents.y * 5f;
+            Vector3 point = handleBounds.center + new Vector3(0, length, 0);
+            switch (finalDirection)
+            {
+                case Direction.x:
+                    point += new Vector3(length / 2f, 0, 0) * subDirection * ((transform.rotation.eulerAngles.y >= 180) ? -1 : 1);
+                    break;
+                case Direction.z:
+                    point += new Vector3(0, 0, length / 2f) * subDirection * ((transform.rotation.eulerAngles.y >= 180) ? -1 : 1);
+                    break;
+            }
+
+            Gizmos.DrawLine(handleBounds.center + new Vector3(0, handleBounds.extents.y, 0), point);
+            Gizmos.DrawSphere(point, length / 8f);
+        }
+
+        protected override void InitRequiredComponents()
+        {
+            InitBody();
+            InitLid();
+            InitHandle();
+        }
+
+        protected override bool DetectSetup()
+        {
+            if (lid == null || body == null || handle == null) return false;
```

Comment:
requires braces here as per contrib guidelines.

keeps things consistent if all code blocks are in braces even if it's a one liner

so

```
{
  return false;
}
```